### DISCUSSION TITLE
[Idea] Show total weight in FO and BO carts

### DISCUSSION THREAD

**Discussion by tjerry** [PrestaShop/PrestaShop#32271] [Ideas]
#### Is your feature request related to a problem?

Prestashop doesn't show total weight in various places in BO and FO.
Namely, in shopping cart (FO), blockcart (FO) and cart details (BO).

When customers order tens or hundreds of products, a current total weight of the cart is really needed.

1) For instance, a customer would like to order many household cleaning products and he wouldn't like to pay for more than 2 parcels (30 kg each).
If shopping cart or blockcart doesn't provide that information (total weight), customers have problems placing such orders. In effect, they unnecessarily call customer service or simply abandon carts and go somewhere else.

2) Another problem is, when customers call and say, "Could you look at my cart, there is something wrong with shipping costs". And without having displayed a total weight in cart details in BO, customer service can't really help. Because, shipping costs often depend on total weight and number of parcels, that an order have to be shipped in.

#### Describe the solution you'd like

A solution is to simply show a total weight in all of those places.
And refresh it in real time (FO) when a cart is updated.

![Weight_in_ajax_cart](https://user-images.githubusercontent.com/2649031/106766561-495e8680-663a-11eb-9fb1-f8e9ab86860a.png)
![Weight_in_shopping_cart](https://user-images.githubusercontent.com/2649031/106766667-5e3b1a00-663a-11eb-840f-6e9d5b8e150c.png)
![Weight_in_bo_cart](https://user-images.githubusercontent.com/2649031/106766682-609d7400-663a-11eb-9f89-11628594ada8.png)

**Comment by hibatallahAouadni** [comment]
Hello @tjerry 

Thanks for the suggestion, the Product Team will take a look.
Please stay tuned !

Ping @PrestaShop/product-team , it's related to #23125, do you think it's useful to add the weight to the product_details in Order or in Shopping Cart BO/FO?

**Comment by tjerry** [comment]
> Ping @PrestaShop/product-team , it's related to #23125, do you think it's useful to add the weight to the product_details in Order or in Shopping Cart BO/FO?

Actually, a better way would be to add a switch in BO settings, to disable or enable it.
I can imagine, there are shops, that wouldn't need it.

**Comment by hibatallahAouadni** [comment]
@tjerry that's a good idea :clap:

**Comment by ruchus** [comment]
Hello @tjerry

I think also that's a good idea.
In fact, i need some times the same as you. More exactly, for my needs it would be nice to show the weight when it's 15kg and more.

Then,  a better way would be to add a switch in BO settings plus a minimum wight value in order to show the weight in BO only when the weight is higher.

**Comment by RogerAndersson** [comment]
I really like this improvement. I have the same need.

**Comment by HartLarsson** [comment]
i thunk is a "must have feature". Will be great to be implemented and leave in BO the opportunity to show or hide the info by a switch.

**Comment by marcos-ba** [comment]
very usefull. i'm waiting for thid option
